services/plugins/plugins/parsers.go

```diff
@@ -81,9 +83,24 @@ func (plugin *Plugin) persistFileSha(destPlugin *Plugin) error {
 	return nil
 }
 
-func (plugin *Plugin) PostCreateProcess(_ *provider.Client, destPluginRaw any) error {
+func (plugin *Plugin) PostCreateProcess(client *provider.Client, destPluginRaw any) error {
 	createdPlugin := destPluginRaw.(*Plugin)
-	plugin.persistFilePath(createdPlugin)
+
+	metadata, err := GetPluginMetadata(createdPlugin.ID, client)
+	if err != nil {
+		return plugin.persistFileSha(createdPlugin)
+	}
+
+	startTime := time.Now()
+
+	for metadata.Status != "ACTIVE" && time.Since(startTime) < 3*time.Minute {
+		fmt.Println("Checking plugin status...")
+		metadata, err = GetPluginMetadata(createdPlugin.ID, client)
```

Comment:
shouldn't we add a wait period (e.g.: of 1s) between each call so it doesn't flood the API?